src/instructlab/configuration.py

```diff
@@ -1207,6 +1208,57 @@ def recreate_train_profiles(overwrite: bool = False) -> bool:
     return fresh_install
 
 
+# read_and_create_system_profiles walks the given dir, reads the templated files and writes them into
+# the DEFAULTS.SYSTEM_PROFILE_DIR preserving their arch, processor name, and yaml file name
+# pylint: disable=broad-exception-caught
+def read_and_create_system_profiles(profiles_dir: str, overwrite: bool) -> bool:
+    fresh_install = False
+    for dirpath, _dirnames, filenames in os.walk(profiles_dir):
+        for filename in filenames:
+            # open file on disk
+            file_path = os.path.join(dirpath, filename)
+            try:
+                # read system profile from disk
+                with open(file_path, "r", encoding="utf-8") as file:
+                    content = yaml.load(file)
+                # ensure that the profile is a valid cfg before
+                try:
+                    _ = Config(**content)
+                except Exception as exc:
+                    raise ValueError(
+                        f"System Profile: {file_path} is not a valid config {exc}"
+                    ) from exc
+                # get just the arch/name/profile_name
+                relative_path = os.path.relpath(file_path, profiles_dir)
+                # new file is the DEFAULTS.SYSTEM_PROFILE_DIR/arch/name/profile_name.yaml
+                new_file = os.path.join(DEFAULTS.SYSTEM_PROFILE_DIR, relative_path)
+                fresh_install = os.path.exists(new_file)
```

Comment:
should this be:
```suggestion
                fresh_install = not os.path.exists(new_file)
```
`fresh_install` should probably be false if the file exists already

Reply:
good catch, thank you